lib/puppet/defaults.rb

```diff
@@ -1598,11 +1599,11 @@ def self.default_vendormoduledir
           it with the `--no-client` option. #{AS_DURATION}",
     },
     :runtimeout => {
-      :default  => 0,
+      :default  => "1h",
       :type     => :duration,
       :desc     => "The maximum amount of time an agent run is allowed to take.
           A Puppet agent run that exceeds this timeout will be aborted.
-          Defaults to 0, which is unlimited. #{AS_DURATION}",
+          Defaults to 1 hour. #{AS_DURATION}",
```

Comment:
Might be worth noting that setting it to 0 means "unlimited", since that's no longer in the docs now that the default has been changed.